Userland/Libraries/LibCrypto/BigInt/UnsignedBigInteger.cpp

```diff
@@ -602,6 +602,151 @@ bool UnsignedBigInteger::operator>=(UnsignedBigInteger const& other) const
     return *this > other || *this == other;
 }
 
+UnsignedBigInteger::CompareResult UnsignedBigInteger::compare_to_double(double value) const
```

Comment:
I don't think there's a need to duplicate all of this logic, just deal with mismatched signs in `SignedBigInteger::compare_to_double` then call `UnsignedBigInteger::compare_to_double` from there, but remove the actual compare steps.

Reply:
Done